Jenkinsfile

```diff
@@ -11,12 +11,19 @@ pipeline {
                 buildDescription "${params.CODENAME} ${params.SRC_BRANCH}"
             }
         }
+        stage('Checkout config') {
+            steps {
+                git branch: "${params.CONFIG_BRANCH}",
+                    credentials-id: "github-ssh-ro",
+                    url: "git@github.com:endlessm/endless-key-config.git"
```

Comment:
I think in this case you're going to need to use `dir` block or it will try to clone the repo to the workspace root. Like:

```
dir('endless-key-config') {
    git ...
}
```

Alternatively you can use the more advanced `checkout` step and a `RelativeTargetDirectory` extension, but [https://www.jenkins.io/doc/pipeline/steps/params/scmgit/](the documentation) clearly states that you shouldn't do that in a pipeline.

Reply:
Are you sure about that? My understanding was that every command is being run inside the directory where endless-key-builder is checked-out, since that is clearly the case for the other steps. So I was expecting this to behave similarly to running `git clone $URL` and checkout the repo in a sub-directory. Perhaps the git step does not behave like `git clone` and specifically places the checkout in the workspace root?

Reply:
(if there is an easy way to experiment with this, I would be happy to try it)

Reply:
I'm pretty sure that's the case, but I created https://ci.endlessos.org/job/test-key-builder/ for you to try things out. I suppose you'll want to edit the parameters manually and use `SRC_BRANCH` set to `T34262`.

Reply:
You were right...

Reply:
But hey, at least playing in https://ci.endlessos.org/job/test-key-builder/ caught the `credentialsId` syntax error :)